analysis/node-wrapper/install.js

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env node
+
+const fs = require("fs");
+const path = require("path");
+
+(async () => {
+    if (process.env.npm_package_name === "codecharta-analysis") {
+		const bins = ["ccsh.ps1", "ccsh.cmd"];
+		const globalOrLocal = process.env.npm_config_global === "true";
+		const scriptLocation = globalOrLocal ? "binScripts/global" : "binScripts/local";
+		const scriptTarget = globalOrLocal
+			? process.env.npm_config_global_prefix
+			: path.join(process.env.npm_config_local_prefix, "node_modules/.bin");
+		var deletionPromises = [];
+
+		bins.forEach(file => {
+			deletionPromises.push(
+				new Promise((res, rej) => {
+					try {
+						fs.copyFile(path.join(scriptLocation, file), path.join(scriptTarget, file), err => {
+							if (err) throw err;
+							console.log(`${file} was moved`);
+							res();
+						})
+					} catch (err) {
+						console.error(err);
+						rej(err);
+					}
+				})
+			)
+		})
+		await Promise.allSettled(deletionPromises);
```

Comment:
```suggestion
    await Promise.allSettled(bins.map(asnyc (file) => {
      await fs.promises.copyFile(path.join(scriptLocation, file), path.join(scriptTarget, file));
      console.log(`${file} was moved`);
    }));
```

Reply:
Files that were not moved are ignored?

Reply:
I'm not sure why, but the catch did not catch the exception and the process was terminated correctly as it is intended. It seems like the callback was not affected by the catch. During testing the install process would be terminated if there was an error in the file copy call, which is the behavior I wanted for the script. 

Do you have a suggestion for this segment, that includes an error throw?